### PR TITLE
Support JDK 16 and 17

### PR DESCRIPTION
**Pull request by Stephan202**
This is a (likely incomplete) set of changes to support Java 16 and 17. These changes follow the approach taken in #56.

### CODE REVIEW

File: newrelic-weaver/src/main/java/com/newrelic/weave/utils/SynchronizedAnnotationNode.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
  * pretty strongly to an ASM version due to the fact that we had to override constructors in the original asm class. If
  * these constructors change or other accept() methods are added we will need to update accordingly.
  *
- * NOTE: If you are upgrading from org.ow2.asm:asm:8.0 / org.ow2.asm:asm-tree:8.0 make sure to double
+ * NOTE: If you are upgrading from org.ow2.asm:asm:9.2 / org.ow2.asm:asm-tree:9.2 make sure to double
```
Comment (Stephan202) [inline newrelic-weaver/src/main/java/com/newrelic/weave/utils/SynchronizedAnnotationNode.java:23]: I checked out the ASM source code to `/tmp/asm` and used the following two commands to independently verify that no further changes to these classes are necessary:
1. A diff of all AWS classes for which New Relic ships with `Synchronized*` subclass:
   ```sh
    git grep -l 'If you are upgrading from' \
      | sed 's,.*Synchronized,asm-tree/src/main/java/org/objectweb/asm/tree/,' \
      | xargs git -C /tmp/asm diff ASM_8_0...ASM_9_2 --
    ```
2. A diff of all `org.objectweb.asm.tree` classes:
    ```sh
    git -C /tmp/asm diff ASM_8_0...ASM_9_2 -- asm-tree/src/main/java/org/objectweb/asm/tree
    ```

**Review by Stephan202 — COMMENTED** [review]
I don't know much about the New Relic Java Agent, so it's likely that further changes to this PR are necessary. For example, perhaps Scala should also be [upgraded](https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html) to 2.13.6+.

I filed this draft PR primarily to further #215. Feel free to copy any or all of these changes. If you'd like to make changes to this branch I can invite you as an outside collaborator to the repository; just let me know.

**Comment by Stephan202** [comment]
Somehow I completely forgot about #366 when I worked on this PR. Sorry about that, @tbradellis. Looking at that PR I notice I forgot one thing: the ASM API level bump; added that now.

**Comment by jasonjkeller** [comment]
@Stephan202 Thanks for this! We're actively working on the [java-16](https://github.com/newrelic/newrelic-java-agent/pull/366) branch that you pointed out above. 

We're going to keep the changes specific to Java 16 support for now to keep our scope focused. Java 16 is a big effort for us as the agent uses quite a bit of reflection internally and, as Java 16 now denies access to those APIs, a lot of things break. Once this sorted out, though, we're optimistic that it will be a quick follow up to support Java 17 once it is officially released.

**Comment by jasonjkeller** [comment]
Going to close this PR in favor of https://github.com/newrelic/newrelic-java-agent/pull/366

**Pull request closed without merging**